### PR TITLE
Improve cloudfront distribution behavior

### PR DESCRIPTION
**Pull request by JonRoma**
Most of these changes are for support of HTTP basic auth in the submodule.

*  Eliminate remaining vestiges of m4, replacing this with a single `basic-auth/regions.tf.json` file that contains all the necessary region data to be used by Terraform at runtime.

*  Add `basic-auth/make-regions` Python script, and `basic-auth/Makefile`. Working together, these use the Amazon CLI to generate a list of AWS regions, and to build a Terraform local variable definition "known_regions" that consists of a list of known region names.

*  Eliminate m4 files, and eliminate code in basic-auth/Makefile previously used to run m4 to generate tf files.

* Add logic to accept region variable in the form

  `regions = ["*"]`

    which specifies that replicas are to be deployed in all supported (and opted-in) regions.

*   Fix bug in `for_each` clause for ACM certificate handling.

*   Refactor DynamoDB code to eliminate the need to create a `aws_dynamodb_table` in each replica region, and to eliminate use of the obsolete `aws_dynamodb_global_table` resource. This function  is now performed using a replica block inside `aws_dynamodb_table`, which can be created as a dynamic block in Terraform.

    This refactoring allows support for DynamoDB global tables V2 (version 2019.11.21).

*   Remove obsolete regional dependency in Makefile for 'terraform validate'.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,7 +13,9 @@ all: tfc test
 tfc: .terraform
 	@# Basic Terraform validation and formating checks
 	terraform version
-	AWS_DEFAULT_REGION=us-east-2 terraform validate
+	terraform init -backend=false
+#	AWS_DEFAULT_REGION=us-east-2 terraform validate
```
Comment (ddriddle) [inline Makefile]: You may want to remove this comment:

```suggestion
```

File: basic-auth/main.tf [hunk 2]
```diff
@@ -1,33 +1,28 @@
-# Autogenerated by main.m4. DO NOT EDIT.
-#
 # https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/globaltables_reqs_bestpractices.html
```
Comment (ddriddle) [inline basic-auth/main.tf]: You should update the link to point to the V2 instructions:

```suggestion
# https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/globaltables.V2.html
```

File: outputs.tf [hunk 3]
```diff
@@ -30,5 +30,13 @@ output "policy_arn" {
...
+  value       = module.basic-auth.dynamodb_table.name != null ? module.basic-auth.dynamodb_table.name : null
 }
+
+#output "regions" {
+#  value = module.basic-auth.regions
+# }
+#
+#output "replica_regions" {
+# value = module.basic-auth.replica_regions
+#}
```
Comment (ddriddle) [inline outputs.tf:42]: I recommend cleaning up dead code:

```suggestion
```

**Review by ddriddle — APPROVED** [review]
@JonRoma I think you may be able to remove the `regions.tf.json` file and dynamically pull available regions using the following Terraform data resource:

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/regions

File: basic-auth/main.tf (hunk 2 above)
Comment (JonRoma) [reply basic-auth/main.tf]: Fixed.

File: outputs.tf (hunk 3 above)
Comment (JonRoma) [reply outputs.tf:42]: This is debugging code, not dead code. :-)

File: Makefile (hunk 1 above)
Comment (JonRoma) [reply Makefile]: Removed.

**Comment by JonRoma** [comment]
@ddriddle, I decided to leave the code as is, but created issue #31 with your suggestion to review later.

**Pull request merged**